File: server/src/main/java/au/com/codeka/warworlds/server/admin/handlers/ChatHandler.java
package au.com.codeka.warworlds.server.admin.handlers;

import com.google.common.collect.ImmutableMap;

import java.util.ArrayList;
import java.util.List;

import au.com.codeka.warworlds.common.proto.Empire;
import au.com.codeka.warworlds.server.handlers.RequestException;
import au.com.codeka.warworlds.server.store.DataStore;
import au.com.codeka.warworlds.server.world.EmpireManager;
import au.com.codeka.warworlds.server.world.WatchableObject;

/**
 * Handles chat messages and stuff.
 */
public class ChatHandler extends AdminHandler {
  @Override
  public void get() throws RequestException {
    List<Long> empireIds = DataStore.i.empires().search(null);
    List<Empire> empires = new ArrayList<>();

    for (Long id : empireIds) {
      WatchableObject<Empire> empire = EmpireManager.i.getEmpire(id);
      if (empire != null) {
        empires.add(empire.get());
      }
    }

    render("chat/index.html", ImmutableMap.<String, Object>builder()
        .put("empires", empires)
        .build());
  }
}

File: server/src/main/kotlin/au/com/codeka/warworlds/server/admin/handlers/ChatHandler.kt
package au.com.codeka.warworlds.server.admin.handlers

import au.com.codeka.warworlds.common.proto.Empire
import au.com.codeka.warworlds.server.store.DataStore
import au.com.codeka.warworlds.server.world.EmpireManager
import com.google.common.collect.ImmutableMap
import java.util.*

/** Handles chat messages and stuff. */
class ChatHandler : AdminHandler() {
  public override fun get() {
    val empireIds = DataStore.i.empires().search(null)
    val empires = ArrayList<Empire>()
    for (id in empireIds) {
      val empire = EmpireManager.i.getEmpire(id)
      if (empire != null) {
        empires.add(empire.get())
      }
    }
    render("chat/index.html", ImmutableMap.builder<String, Any>()
        .put("empires", empires)
        .build())
  }
}
